Initial Documentation

DominikPinsel opened the pull request:
I started working on our documentation. Currently there are our diagrams and most the content of the power point document. I'd suggest we try to get this incomplete version of the documentation on the main branch and fill it then with targeted pull requests, which document only one topic.
Like this we are able to discuss and improve each topic separately. A multi purpose documentation pull request is probably not a good idea.

Things we should still discuss are
- documentation structure
- content of the documentation already added

<sub>Dominik Pinsel <dominik.pinsel@daimler.com>, Daimler TSS GmbH, [legal info/Impressum](https://github.com/Daimler/daimler-foss/blob/master/LEGAL_IMPRINT.md)</sub>

DominikPinsel commented:
I removed some of the documentation text. Like this it should be less prone to dicsussion and can be merged more easily.

alexandrudanciu reviewed (commented):
The following changes and improvements are suggested:
- Replace the section name “Preface” with “Overview”.
- Add top level section “Core Framework”. Move sections “Contracts” and “Data Transfer” as subsections below “Core Framework”.
- Add top level section “Development of Extensions”. Instead of the generic term “Use Cases”, add more specific section names below “Development of Extensions”, e.g., “Data Offering”, “Contract Templating”, etc.
- Add top level sections “Installation”, “Configuration”, and “Troubleshooting”.
- The document Agenda.md seems to be more related to the development process than to the documentation of the result and should be moved.

DominikPinsel commented:
> The following changes and improvements are suggested:
> 
>     * Replace the section name “Preface” with “Overview”.
> 
>     * Add top level section “Core Framework”. Move sections “Contracts” and “Data Transfer” as subsections below “Core Framework”.
> 
>     * Add top level section “Development of Extensions”. Instead of the generic term “Use Cases”, add more specific section names below “Development of Extensions”, e.g., “Data Offering”, “Contract Templating”, etc.
> 
>     * Add top level sections “Installation”, “Configuration”, and “Troubleshooting”.
> 
>     * The document Agenda.md seems to be more related to the development process than to the documentation of the result and should be moved.

Hello Alexandru,

thanks a lot for your feedback. I made the following adjustments

> * Replace the section name “Preface” with “Overview”.

That sounds really better. Done.

> * Add top level section “Core Framework”. Move sections “Contracts” and “Data Transfer” as subsections below “Core Framework”

Great idea to document all the core components. I named the top level section "Core Extensions".  I hope this name is also fine.

> * Add top level section “Development of Extensions”. Instead of the generic term “Use Cases”, add more specific section names below “Development of Extensions”, e.g., “Data Offering”, “Contract Templating”, etc.

You are correct that we need this documented, but that's not what I meant when I added use cases. I see now that the section there is probably just confusing and removed it. Hope this is also fine.

> * Add top level sections “Installation”, “Configuration”, and “Troubleshooting”.

Yea, we are missing these. Done.

> * The document Agenda.md seems to be more related to the development process than to the documentation of the result and should be moved.

Yea. This is probably not something we want on the main branch. I removed the file and will make issues in the repository here for the stuff mentioned there.

Are you fine with this?

alexandrudanciu reviewed (commented):
The following improvements are proposed:

The structure of the sections on the architecture could be perceived as inconsistent:
- In Readme.md: Section “Architecture” consisting of subsection “Core Extensions”
- In Architecture.md: Section “Architecture” consisting of subsection “Runtime and Extensions”
A solution would be to move “Core Extensions” in the Readme.md one level up (which maybe was already intended).

Currently, we are using both the terms “Core Extension”  and “Extension”. This could suggest, there are multiple types of extensions (core and non-core). I propose the following two alternatives:
1. To only refer to “Extensions” (which implies that the existing core is extended).
2. To distinguish between “Core Extensions” and “IDS Extensions”, as there is a dedicated IDS SPI layer.

The section “Installation” is missing. If it was intended, then it probably was forgotten.

DominikPinsel commented:
> The following improvements are proposed:
> 
> The structure of the sections on the architecture could be perceived as inconsistent:
> 
> * In Readme.md: Section “Architecture” consisting of subsection “Core Extensions”
> * In Architecture.md: Section “Architecture” consisting of subsection “Runtime and Extensions”
>   A solution would be to move “Core Extensions” in the Readme.md one level up (which maybe was already intended).
> 
> Currently, we are using both the terms “Core Extension” and “Extension”. This could suggest, there are multiple types of extensions (core and non-core). I propose the following two alternatives:
> 
> 1. To only refer to “Extensions” (which implies that the existing core is extended).
> 2. To distinguish between “Core Extensions” and “IDS Extensions”, as there is a dedicated IDS SPI layer.
> 
> The section “Installation” is missing. If it was intended, then it probably was forgotten.

@alexandrudanciu 

You are right, I totally forgot the installation section. I've added it to the README.md.

Now that you've said it I also see that there is not something like a core extension. I just assumed it because there are some "more important" extensions in the core folder. But it's definatelly wrong to just introduce the term like this.
As I think there are still some extension discussions still ongoing, maybe it's again an idea to just remove these sections for now from the README.md (Core Extensions, Contracts & Data Transfer). I also removed the core-extension and how-to directory.

alexandrudanciu reviewed (approved):
Proposed improvements addressed.

MoritzKeppler commented:
think we should add a disclaimer like this "docs is work in progress, all content reflect the current state of discussion, not final decisions" to the README.md under docs

MoritzKeppler commented on docs/README.md:
```diff
@@ -0,0 +1,33 @@
+**Please note**
+### This document reflects work in progress
+All content reflects the current state of discussion, not final discussions.
```
```suggestion
### This document is work in progress
All content reflects the current state of discussion, not final decisions.
```

MoritzKeppler commented on docs/README.md:
```diff
@@ -0,0 +1,33 @@
+**Please note**
+### Note: work in progress
```
```suggestion
### Work in progress
```
